plugins/ipfs/docker/dockeripfs.go

```diff
@@ -0,0 +1,547 @@
+package plugindockeripfs
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"path/filepath"
+
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/gxed/errors"
+	"github.com/ipfs/go-cid"
+	config "github.com/ipfs/go-ipfs/repo/config"
+	serial "github.com/ipfs/go-ipfs/repo/fsrepo/serialize"
+	"github.com/multiformats/go-multiaddr"
+	cnet "github.com/whyrusleeping/go-ctrlnet"
+
+	"github.com/ipfs/iptb/plugins/ipfs"
+	"github.com/ipfs/iptb/testbed/interfaces"
+	"github.com/ipfs/iptb/util"
+)
+
+var ErrTimeout = errors.New("timeout")
+
+var PluginName = "dockeripfs"
+
+var NewNode testbedi.NewNodeFunc
+var GetAttrDesc testbedi.GetAttrDescFunc
+var GetAttrList testbedi.GetAttrListFunc
+
+const (
+	attrIfName = "ifname"
+)
+
+func init() {
+	NewNode = func(dir string, extras map[string]interface{}) (testbedi.TestbedNode, error) {
```

Comment:
You don't have to assign these like this, it should work the same if you just define it as:
```
func NewNode(dir string, extras map[string]interface{}) (testbedi.TestbedNode, error) {
...
...
}
```